public/app/themes/justice/src/patch/js/ccfw-storage.js

```diff
@@ -1,29 +1,82 @@
 /**
  * A script for determining our storage method
  *
- * Falls back to cookies if localStorage is unavailable
+ * Falls back to cookies if localStorage is unavailable, otherwise
+ * we disable storage all together.
  */
-if (window.localStorage === undefined) {
-    Storage = {
-        getItem: (key) => {
-            const value = `; ${document.cookie}`;
-            const parts = value.split(`; ${key}=`);
+let Storage = {
+    disabled: true
+}
+
+const CCFWStorage = {
+    /**
+     * Both local() and cookie() return booleans
+     * false = invalid
+     * true = valid
+     * First we test for localStorage, if this is false, we test for cookies
+     * If cookie returns true, valid() returns true, otherwise, valid() returns
+     * false.
+     *
+     * @returns {boolean}
+     */
+    valid: () => {
+        return CCFWStorage.local() || CCFWStorage.cookie()
+    },
+    local: () => {
+        let mod = 'ccfw';
+        try {
+            localStorage.setItem(mod, mod);
+            localStorage.removeItem(mod);
+            return true;
+        } catch (e) {
+            return false
+        }
+    },
+    cookie: () => {
+        try {
+            // try and set it...
+            document.cookie = 'cookie_test=1'
+            // test it exists...
+            const cookiesEnabled = document.cookie.indexOf('cookie_test=') !== -1
+            // remove it...
+            document.cookie = 'cookie_test=1; expires=Thu, 01-Jan-1970 00:00:01 GMT'
+            return cookiesEnabled
```

Comment:
This is an odd one...

> ... in some in situations `document.cookie` is exposed but throws a
> SecurityError if you try to access it...

https://github.com/Modernizr/Modernizr/blob/master/feature-detects/storage/cookies.js

We checked the try/catch result in case no error was thrown but we still couldn't write the cookie.